designs/2018-processors-improvements/README.md

```diff
@@ -0,0 +1,202 @@
+
+End-users can also mix and match processors, such as in this example:
+
+```js
+module.exports = {
+    plugins: ["markdown", "vue", "html"],
+    parserOptions: {
+        ecmaVersion: 2018
+    },
+    overrides: [
+        {
+            files: ["*.md"],
+            processors: ["markdown/markdown", "vue/vue"]
+        },
+        {
+            files: ["*.htm", "*.html"],
+            processors: ["html/html", "vue/vue"]
+        }
+    ]
+};
+```
+
+This configuration uses the same `vue` named processor from `eslint-plugin-vue` for both HTML and Markdown files, but only after an appropriate named processor is used to extract the Vue code.
+
+## Documentation
+
+The following areas of the website will need to be updated:
+
+* [Processors in Plugins](https://eslint.org/docs/developer-guide/working-with-plugins#processors-in-plugins)
+* [Configuring ESLint](https://eslint.org/docs/user-guide/configuring#configuring-eslint)
+
+Additionally, a blog post announcement should be made both to publicize the change and to encourage plugin owners to update their processors to be named instead of extension-named.
+
+## Drawbacks
+
+There are some drawbacks to this proposal, namely:
+
+1. Existing plugins would have to be updated so they can be used in this way.
+1. Existing configurations would have to be updated to make use of this design.
+
+Both of these drawbacks create a barrier to adoption and require changes by the end-user in order to take advantage of this design.
+
+## Backwards Compatibility Analysis
+
+This proposal is 100% backwards compatible until we remove the old way of defining processors. Both named and extension-based processors can be define in the same plugin, such as:
+
+```js
+const processor = {
+    preprocess() {},
+    postprocess() {}
+};
+
+module.exports = {
+    processors: {
+        ".md": processor,
+        markdown: processor
+    }
+};
+```
+
+We can continue to honor the extension-based processors by converting the definition into an entry in the referencing configuration's `overrides` section. So, for the existing extension-based Markdown processor, we would internally do this:
+
+```js
+module.exports = {
+    plugins: ["markdown"],
+    parserOptions: {
+        ecmaVersion: 2018
+    },
+    overrides: [
+
+        // this entry is added internally for backwards compatibility
+        {
+            files: ["*.md"],
+            processors: ["markdown/markdown"]
+        },
+
+        // any already-existing overrides would go here
+    ]
+};
+```
+
+Making this extra `overrides` entry internally would eliminate a special case when determining which processors to use for which files.
+
+At a future point in time, we can determine whether or not continue with this backwards-compatible behavior.
+
+## Alternatives
+
+@mysticatea submitted an alternative design proposal: [rfs/#1](https://github.com/eslint/rfcs/pulls/1)
```

Comment:
Do we want to have a compare/contrast note here (or in the other proposal)?

Maybe this raises a more general question: Where is the best place to discuss or debate alternative proposals for the same area/feature?

Reply:
Yes, I will do that in the next revision.